From 0b63152ae5d572c4a817a53210d8907f3dd11bd7 Mon Sep 17 00:00:00 2001
From: Simon Hausmann <simon.hausmann@nokia.com>
Date: Mon, 25 Jun 2012 09:47:34 +0200
Subject: [PATCH 05/11] Fix build on linux-i386 where gcc would produce
 warnings

It is the intention to _not_ use -Werror for production builds. Unfortunately
the qmake expression had an error that made the -Werror suppression only work
on x86-64 but not on i386. A surrounding if() as suggested by Ossi fixes this.

In trunk this is already fixed differently by the use of an extra scope.

Task: QTBUG-23987
---
 Source/WebKit.pri | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/WebKit.pri b/Source/WebKit.pri
index 5080119..ae7b99c 100644
--- a/Source/WebKit.pri
+++ b/Source/WebKit.pri
@@ -102,7 +102,7 @@ CONFIG -= warn_on
 
 # Treat warnings as errors on x86/Linux/GCC
 linux-g++* {
-    !CONFIG(standalone_package):isEqual(QT_ARCH,x86_64)|isEqual(QT_ARCH,i386): QMAKE_CXXFLAGS += -Werror
+    !CONFIG(standalone_package):if(isEqual(QT_ARCH,x86_64)|isEqual(QT_ARCH,i386)): QMAKE_CXXFLAGS += -Werror
 
     greaterThan(QT_GCC_MAJOR_VERSION, 3):greaterThan(QT_GCC_MINOR_VERSION, 5) {
         if (!contains(QMAKE_CXXFLAGS, -std=c++0x) && !contains(QMAKE_CXXFLAGS, -std=gnu++0x)) {
-- 
1.8.0.2

